WordPress/Classes/ViewRelated/Aztec/New Group/AztecVerificationPromptHelper.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  AztecVerificationPromptHelper.swift
+//  WordPress
+//
+//  Created by Jan Klausa on 05.10.17.
+//  Copyright © 2017 WordPress. All rights reserved.
+//
+
+import UIKit
+
+internal class AztecVerificationPromptHelper: NSObject {
+
+    private let managedObjectContext: NSManagedObjectContext
+    private let accountService: AccountService
+    private let wpComAccount: WPAccount
+
+    private weak var displayedAlert: FancyAlertViewController?
+    private var completionBlock: AztecVerificationPromptCompletion?
+
+    typealias AztecVerificationPromptCompletion = (Bool) -> ()
+
+    init?(managedObjectContext: NSManagedObjectContext, for account: WPAccount?) {
+        self.managedObjectContext = managedObjectContext
+        self.accountService = AccountService(managedObjectContext: managedObjectContext)
+
+        guard let wpComAccount = self.accountService.defaultWordPressComAccount(),
+                  wpComAccount == account,
```

Comment:
Thinking a bit about this... 
I think this is fine if `wpComAccount` and `account` belong to the same `NSManagedObjectContext` but ... since a different NSManagedObject context could be passed as a parameter to this method, its possible for the `==` to fail if the objects belong to different contexts, even if they point to the same record.

As a best practice, we try to make sure objects belong to the same context when performing a comparison or we try to compare NSManagedObjectID since those will be the same regardless of context.

Reply:
That's a grrreat point! I'm mostly used to operating in single-MOC (except for few background tasks like importing/sync) environments, so this is a bit of a blind spot for me. 
I managed to rewrite that to avoid that check in the first place anyway, but thanks for reminding me of this :)